### DISCUSSION TITLE
Looking for: IRL Testers

### DISCUSSION THREAD

**Discussion by MadWombat** [RonSijm/ButtFish#37] [General]
I am not sure how to do testing on this product. If there are unit tests, how do you mock a butt? If there are end to end tests, what do you use for backend?

**Comment by zane0x** [comment]
LOL，Need volunteers to contribute their butt

**Comment by RonSijm** [comment]
Hey - as of now there are no tests yet. The code is build with that in mind for the future though.

How that would work: [The DeviceConnector.cs](https://github.com/RonSijm/ButtFish/blob/ac4a34a22cbe11ad333c2ac45fdf80fd0e3057b4/RonSijm.ButtFish/Connectors/DeviceConnector.cs)  would get it's connectors abstracted, so the constructor would just get an `IEnumerable<IDeviceConnector> connectors` parameter, (which would either be [YeelightConnector](https://github.com/RonSijm/ButtFish/blob/ac4a34a22cbe11ad333c2ac45fdf80fd0e3057b4/RonSijm.ButtFish/Connectors/YeelightConnector.cs), [ButtplugConnector](https://github.com/RonSijm/ButtFish/blob/ac4a34a22cbe11ad333c2ac45fdf80fd0e3057b4/RonSijm.ButtFish/Connectors/ButtplugConnector.cs) - Or a new MockConnector

From there I'd use [Moq](https://github.com/moq/moq4) and return a new `Mock<IDeviceAbstraction>()` to validate.

I haven't had time (or purpose) to create that so far though

**Comment by citypaul** [comment]
Hmm, this might be a project that requires a large amount of manual testing. Any volunteers?

**Comment by anonymousspacething** [comment]
The only thing missing here is finding someone who both knows how to play chess, and also has a wireless buttplug that is compatible with buttplug.io

I unfortunately do not know how to play chess.
